### DISCUSSION TITLE
Prevent scroll to top on render

### DISCUSSION THREAD

**Discussion by Flobbos** [livewire/livewire#3410] [1. Help] [answered]
Hello,

I have been plagued by this issue for quite some time now so I thought I'd describe my problem here. The issue is that I have a fairly complex page creation component that lets the user add content objects to a page. Everything works fine but when the content gets larger and you need to scroll down to add more stuff, the page scrolls to the top every time a new element is being added. Each element is a Livewire component in itself because they all have different input options (text, photo, video etc.). 

Through trial and error I have figured out that adding regular HTML elements does not trigger this behaviour but as soon as a Livewire component is added to the DOM the browser scrolls to the top. Why? 

I could technically put EVERYTHING into the same component but that would make it very tough to maintain. I would very much appreciate some insight into why this is happening and how to possibly prevent it.

**Comment by Flobbos** [comment, marked as answer]
After poking around for a bit I figured out that I needed to wrap each livewire in a separate div.

**Reply by jhonoryza** [reply]
can u give example please :D, i have same issue like you

**Reply by theChaosCoder** [reply]
Basically try adding a div around your component / element `<div> ... </div>`. In a for loop use key
More info here: https://laravel-livewire.com/docs/2.x/troubleshooting
